### DISCUSSION TITLE
Continues to work with screen off?

### DISCUSSION THREAD

**Discussion by brianjmurrell** [OpenTracksApp/OpenTracks#1154] [Q&A] [answered]
If I start a recording and don't set the option to keep the screen on and then put my phone in my pocket will OpenTracks continue to sample the location data and sensor data for the duration of my activity?

What prevents Android from thinking the app is idle/not-being-used and killing it as it seems to so aggressively do with apps in later Android releases -- in the name of battery efficiency?

**Comment by dennisguse** [comment, marked as answer]
OpenTracks uses a foreground service and a CPU wake lock.
Some phones ignore this and then battery optimization must be disabled as well.
Huawei is problematic...

https://developer.android.com/guide/components/foreground-services
